sdc_client/balancer.py

```diff
@@ -12,12 +13,61 @@ class StreamsetsBalancer:
     def __init__(self):
         self.streamsets_pipelines: Dict[IStreamSets, List[IPipeline]] = get_streamsets_pipelines()
 
+    def _refresh_object(self):
+        self._preferred_types = set([ss.get_preferred_type() for ss in self.streamsets_pipelines if ss.get_preferred_type()])
```

Comment:
How about set comprehension
`{ss.get_preferred_type() for ss in self.streamsets_pipelines if ss.get_preferred_type()}`